roles/matrix-synapse/defaults/main.yml

```diff
@@ -27,6 +27,8 @@ matrix_synapse_storage_path: "{{ matrix_synapse_base_path }}/storage"
 matrix_synapse_media_store_path: "{{ matrix_synapse_storage_path }}/media-store"
 matrix_synapse_ext_path: "{{ matrix_synapse_base_path }}/ext"
 
+ matrix_synapse_container_default_federation_api_port: 8448
```

Comment:
Using `_default` in the name doesn't make much sense.

`matrix_synapse_container_federation_api_port` seems better.

This variable can default to `8448` here in the `matrix-synapse` role, and can be redefined in terms of the `matrix_federation_public_port` variable via `group_vars/matrix_servers` (although it's not really necessary to make the public port and the internal port match).

Reply:
Will change it to `matrix_synapse_container_federation_api_port`. Is there any naming convention that you follow ? Also what is the better place to discuss development ideas ?

Reply:
Well, in this case, "default" sounds like a default only. It sounds like a value that should be defined and used as a reference without being redefined.

It may make sense to use it like this, actually:

```yaml
# This is some internal variable with a default value.
default_value: 5555

# Users can redefine this.
actual_value: "{{ default_value }}"

something_else: "https://localhost:{{ actual_value }}"
```

------

In your case, you intend to have this variable redefined by users. Redefining `matrix_synapse_container_default_federation_api_port: 55555` sounds strange. You've actually redefined the federation API port and many other variables to use 55555. The default for Synapse is still 8448 (as per spec and natural consensus).

Reply:
I agree. Really appreciate your explanation!